pkg/apis/componentconfig/types.go

```diff
@@ -164,171 +164,219 @@ type GroupResource struct {
 type KubeControllerManagerConfiguration struct {
 	metav1.TypeMeta
 
+	// GenericComponentConfiguration holds configuration for GenericComponent
+	// related features both in cloud controller manager and kube-controller manager.
+	GenericComponent GenericComponentConfiguration
+	// KubeCloudSharedConfiguration holds configuration for shared related features
+	// both in cloud controller manager and kube-controller manager.
+	KubeCloudShared KubeCloudSharedConfiguration
+	// DebuggingConfiguration holds configuration for Debugging related features.
+	Debugging DebuggingConfiguration
+	// CloudProviderConfiguration holds configuration for CloudProvider related features.
+	CloudProvider CloudProviderConfiguration
+	// ServiceControllerConfiguration holds configuration for ServiceController
+	// related features.
+	ServiceController ServiceControllerConfiguration
```

Comment:
let's put an empty line above this first controllers. And then sort the controllers by name, same for the ControllerConfigurations below.